### DISCUSSION TITLE
General requirements for the projects

### DISCUSSION THREAD

**Discussion by dtrendafilov** [SofiaCPP/IPL#114] [Projects]
# The general requirements for the projects

- implemented in C / C++
- instructions for how to build, preferably project files / singe command builds
- tests

## How projects are going to be graded:

- correctness - make sure to squash bugs, crashes, leaks
- readability
- completeness of the implementation

## Project assignments

- Projects are limited to 4 assignees, on first come first serve basis.
- Everyone should have been assigned a project by 24 January 2023 ❗
- Assign yourself for a project by commenting in the project theme - "I will do this project" 

## Other project ideas

If you have an idea for a project, please do suggest that in the _General_ category, until we formulate the idea well enough to be promoted to a project that other people can take as well.

**Comment by PeterBadzhakov** [comment]
Ще правя този проект. Петър Баджаков 4MI3400199
